### PR TITLE
New Guardian Weekly tests

### PR DESCRIPTION
**Pull request by JustinPinner**
This PR adds additional test coverage to the GW promo page to make sure everything is as expected in the run up to the 10th October changeover.

By default the new promo page based tests are excluded from the build process, but can be run locally with `sbt weekly-promo-acceptance-test`

This isn't a definitive set of scenarios but should give us confidence that the right things are happening.

### CODE REVIEW

File: test/acceptance/WeeklyPromoSpec.scala [hunk 1]
```diff
@@ -0,0 +1,350 @@
...
+        1,
+        "US$234 for 1 year, then standard rate (US$260 every year)")
+      )
+
+    }
+
+    // NEW PRICE/PRODUCT TEST
+
+    scenario("UK user lands on Guardian Weekly promotion page AFTER 10:45 on 10/10/18", WeeklyPromoAcceptance) {
+      val promoCode = "/p/10ANNUAL"
```
Comment (jacobwinch) [inline test/acceptance/WeeklyPromoSpec.scala]: We could pick up coverage of [this change](https://github.com/guardian/subscriptions-frontend/pull/1165) by navigating to /weekly and allowing the redirect to choose the promo?

File: test/acceptance/pages/WeeklyPromo.scala [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+      val namedMenu = cssSelector(dropDownSelector).findElement
+
+      namedMenu.exists{ menu =>
+        val elLinkDivs = menu.underlying.findElements(By.tagName("div"))
+        if (elLinkDivs.size() > 0) {
+          val anchors = elLinkDivs.get(0).findElements(By.className("weekly__package"))
+          if (anchors.size() > 0) {
+            val anchor = anchors.get(index)
+            val href = anchor.getAttribute("href")
+            href.endsWith(url)
```
Comment (jacobwinch) [inline test/acceptance/pages/WeeklyPromo.scala]: 😱

File: build.sbt [hunk 3]
```diff
@@ -120,8 +120,9 @@ javaOptions in Universal ++= Seq(
 
 addCommandAlias("devrun", "run  9200")
 addCommandAlias("prodrun", "run 9200")
-addCommandAlias("fast-test", "testOnly -- -l Acceptance")
+addCommandAlias("fast-test", "testOnly -- -l Acceptance -l WeeklyCheckoutAcceptance -l WeeklyPromoAcceptance")
```
Comment (jacobwinch) [inline build.sbt]: 🏅

File: build.sbt [hunk 4]
```diff
@@ -120,8 +120,9 @@ javaOptions in Universal ++= Seq(
...
 addCommandAlias("acceptance-test", "testOnly acceptance.CheckoutSpec")
+addCommandAlias("weekly-promo-acceptance-test", "testOnly acceptance.WeeklyPromoSpec")
```
Comment (jacobwinch) [inline build.sbt]: 👍for adding a new alias for these

File: test/acceptance/WeeklyPromoSpec.scala (hunk 1 above)
Comment (JustinPinner) [reply test/acceptance/WeeklyPromoSpec.scala]: The redirect kills the `gwoct18` param I think. But if not, then yes.

File: test/acceptance/pages/WeeklyPromo.scala (hunk 2 above)
Comment (JustinPinner) [reply test/acceptance/pages/WeeklyPromo.scala]: I know. The URL in the page obvs is the fully qualified thing, protocol, host etc included. That varies from platform to platform :-/

File: test/acceptance/WeeklyPromoSpec.scala (hunk 1 above)
Comment (jacobwinch) [reply test/acceptance/WeeklyPromoSpec.scala]: Feel free to consider investigating/adding this in a separate PR though, the changes here are already providing a lot of value 👍

Comment (JustinPinner) [reply test/acceptance/WeeklyPromoSpec.scala]: Actually it seems to send it through. Looks a little weird but might just work. Let me try...

**Comment by lmath** [comment]
👍 nice work!

**Pull request merged**